File: php-advisor-core/src/Payloads/ClimatologyPayload.php
<?php

namespace StormGeo\AdvisorCore\Payloads;

class ClimatologyPayload extends BasePayload
{
  /**
   * @var int
   */
	public $localeId;

  /**
   * @var string
   */
	public $latitude;

  /**
   * @var string
   */
	public $longitude;

  /**
   * @var string
   */
  public $stationId;

  /**
   * @var array<string>
   */
	public $variables;

  /**
   * @param array{localeId:int,latitude:string,longitude:string,stationId:string,variables:array<string>} $parameters
   */
  public function __construct($parameters = [])
  {
    parent::__construct(
      ['localeId', 'latitude', 'longitude', 'stationId', 'variables'],
      $parameters
    );
  }

  /**
   * @var array
   */
  public function getQueryParams()
  {
    return (array) $this;
  }
}

File: dotnet-advisor-core/Payloads/ClimatologyPayload.cs
namespace StormGeo.AdvisorCore.Payloads;

public class ClimatologyPayload
{
    public int? LocaleId { get; set; }
    public string? Latitude { get; set; }
    public string? Longitude { get; set; }
    public string? StationId { get; set; }
    public string[]? Variables { get; set; }

    public string GetQueryParams()
    {
        return new QueryParameterBuilder()
            .AddLocaleId(LocaleId)
            .AddLatLon(Latitude, Longitude)
            .AddStationId(StationId)
            .AddVariables(Variables)
            .Build();
    }
}
